src/main/java/net/minecraftforge/fml/common/network/FMLIndexedMessageToMessageCodec.java

```diff
@@ -65,11 +65,17 @@ public void handlerAdded(ChannelHandlerContext ctx) throws Exception
     @Override
     protected final void encode(ChannelHandlerContext ctx, A msg, List<Object> out) throws Exception
     {
+        String channel = ctx.channel().attr(NetworkRegistry.FML_CHANNEL).get();
+        Class<?> clazz = msg.getClass();
+        if (!types.containsKey(clazz))
+        {
+            throw new RuntimeException("Undefined discriminator for message type " + clazz.getSimpleName() + " in channel " + channel);
+        }
+        byte discriminator = types.get(clazz);
         PacketBuffer buffer = new PacketBuffer(Unpooled.buffer());
-        byte discriminator = types.get(msg.getClass());
         buffer.writeByte(discriminator);
         encodeInto(ctx, msg, buffer);
-        FMLProxyPacket proxy = new FMLProxyPacket(buffer/*.copy()*/, ctx.channel().attr(NetworkRegistry.FML_CHANNEL).get());
+        FMLProxyPacket proxy = new FMLProxyPacket(buffer/*.copy()*/, channel);
```

Comment:
You should remove `/*.copy()*/`, it looks like it hasn't been active since before 1.7.10.